### PR TITLE
Save clips to s3

### PR DESCRIPTION
**Pull request by ijunaid8989**

### CODE REVIEW

File: lib/evercam_media/snapshot/storage.ex [hunk 1]
```diff
@@ -579,25 +580,23 @@ defmodule EvercamMedia.Snapshot.Storage do
...
-    post_url = "#{@seaweedfs}#{file_path}"
+    file_path = "#{camera_exid}/clips/#{archive_id}/thumb-#{archive_id}.jpg"
 
     "#{path}thumb-#{archive_id}.jpg"
     |> File.open([:read, :binary, :raw], fn(file) -> IO.binread(file, :all) end)
     |> case do
       {:ok, content} ->
-        case HTTPoison.post(post_url, {:multipart, [{file_path, content, []}]}, [], hackney: [pool: :seaweedfs_upload_pool]) do
-          {:ok, _response} -> :noop
+        case ExAws.S3.put_object(@s3_bucket, file_path, content, []) |> ExAws.request do
```
Comment (azharmalik3) [inline lib/evercam_media/snapshot/storage.ex]: Using the following function to save.
https://github.com/evercam/evercam-server/blob/master/lib/evercam_media/timelapse_recording/s3.ex#L37

Also, add opts `[content_type: "image/jpeg"]` for image save request and `[content_type: "video/mp4"]` for video file.

File: lib/evercam_media/snapshot/storage.ex [hunk 2]
```diff
@@ -607,13 +606,12 @@ defmodule EvercamMedia.Snapshot.Storage do
...
       {:ok, %HTTPoison.Response{status_code: 200, body: content}} ->
-        file_path = "/#{camera_exid}/clips/#{archive_id}.#{extension}"
+        file_path = "#{camera_exid}/clips/#{archive_id}/#{archive_id}.#{extension}"
         File.mkdir_p("#{@root_dir}/#{archive_id}/")
         File.write("#{@root_dir}/#{archive_id}/#{archive_id}.#{extension}", content)
-        post_url = "#{@seaweedfs}#{file_path}"
-        case HTTPoison.post(post_url, {:multipart, [{file_path, content, []}]}, [], hackney: [pool: :seaweedfs_upload_pool]) do
-          {:ok, _response} -> :noop
-          {:error, error} -> Logger.info "[save_archive_file] [#{camera_exid}] [#{archive_id}] [#{inspect error}]"
+        case ExAws.S3.put_object(@s3_bucket, file_path, content, []) |> ExAws.request do
```
Comment (azharmalik3) [inline lib/evercam_media/snapshot/storage.ex]: Use same save function.

File: lib/evercam_media/snapshot/storage.ex [hunk 3]
```diff
@@ -629,23 +627,20 @@ defmodule EvercamMedia.Snapshot.Storage do
...
   def seaweedfs_save_video_file(camera_exid, archive_id, content) do
-    hackney = [pool: :seaweedfs_upload_pool]
-    file_path = "/#{camera_exid}/clips/#{archive_id}.mp4"
-    post_url = "#{@seaweedfs}#{file_path}"
-    case HTTPoison.post(post_url, {:multipart, [{file_path, content, []}]}, [], hackney: hackney) do
-      {:ok, _response} -> :noop
-      {:error, error} -> Logger.info "[save_video] [#{camera_exid}] [#{archive_id}] [#{inspect error}]"
+    file_path = "#{camera_exid}/clips/#{archive_id}/#{archive_id}.mp4"
+
+    case ExAws.S3.put_object(@s3_bucket, file_path, content, []) |> ExAws.request do
```
Comment (azharmalik3) [inline lib/evercam_media/snapshot/storage.ex]: Use the same save function.

File: lib/evercam_media/snapshot/storage.ex [hunk 4]
```diff
@@ -579,25 +580,23 @@ defmodule EvercamMedia.Snapshot.Storage do
...
   def load_archive_thumbnail(camera_exid, archive_id) do
-    file_path = "/#{camera_exid}/clips/thumb-#{archive_id}.jpg"
-    get_url = "#{@seaweedfs}#{file_path}"
-    case HTTPoison.get("#{get_url}", [], hackney: [pool: :seaweedfs_download_pool]) do
-      {:ok, %HTTPoison.Response{status_code: 200, body: body}} -> body
-      {:ok, %HTTPoison.Response{status_code: 404}} -> Util.default_thumbnail
-      {:error, %HTTPoison.Error{reason: _reason}} -> Util.default_thumbnail
+    file_path = "#{camera_exid}/clips/#{archive_id}/thumb-#{archive_id}.jpg"
+
+    case ExAws.S3.get_object(@s3_bucket, file_path) |> ExAws.request do
```
Comment (azharmalik3) [inline lib/evercam_media/snapshot/storage.ex]: Use the following function to get the file from s3.
https://github.com/evercam/evercam-server/blob/master/lib/evercam_media/timelapse_recording/s3.ex#L102

File: lib/evercam_media/snapshot/storage.ex (hunk 1 above)
Comment (ijunaid8989) [reply lib/evercam_media/snapshot/storage.ex]: Okay

FYI. this above function doesn't return any tuple to handle it in a `case`, So there will not Error logging.

Comment (azharmalik3) [reply lib/evercam_media/snapshot/storage.ex]: Then add case's in this function.

**Pull request merged**